control_software/scripts/hera_snap_redis_monitor.py

```diff
@@ -275,16 +275,57 @@ def print_ant_log_messages(corr):
         corr.r.hmset('version:%s:%s' % (__package__, os.path.basename(__file__)), {'version':__version__, 'timestamp':datetime.datetime.now().isoformat()})
 
         # Get antenna stats
-        input_stats = corr.do_for_all_f("get_stats", block="input", kwargs={"sum_cores": True})
+        # do_for_all_f breaking on adc stats, casting to None 25 Aug 2020
+        # input_stats = corr.do_for_all_f("get_stats", block="input", kwargs={"sum_cores": True})
+        input_stats = {}
+        histograms = {}
+        bins = np.arange(-128, 129)
+        bin_centers = (bins[1:] + bins[:1]) / 2
+        for feng in corr.fengs:
+            histograms[feng.host] = []
+            input_stats[feng.host] = []
+            for i in range(3):
+                try:
+                    x, y = feng.input.get_adc_snapshot(i)
+                    hist_x, _ = np.histogram(x, bins=bins)
+                    hist_y, _ = np.histogram(x, bins=bins)
+                    histograms[feng.host].append([bin_centers, hist_x])
+                    histograms[feng.host].append([bin_centers, hist_y])
+                    input_stats[feng.host].append(
+                        [
+                            x.mean(), np.mean(x**2), np.sqrt(np.mean(x**2))
+                        ]
+                    )
+                    input_stats[feng.host].append(
+                        [
+                            y.mean(), np.mean(y**2), np.sqrt(np.mean(y**2))
+                        ]
+                    )
+                except:  # noqa
```

Comment:
Capture error and report to logger?

Reply:
Yeah I can pass the error to the logger here. It is usually a TFTP error while calling the "get_adc_snapshot". @jkocz mentioned there may be momentary bandwidth overloads when getting the adc snapshot. Wasn't sure if we wanted to pump all the TFTP errors into the logger but can't hurt to have them.